### ISSUE TITLE
Ending base URI with a slash results in double slash in produced URIs

### ISSUE THREAD

**Issue by a1exsh** [dehora/nakadi-java#330]
It looks like the following part of code is responsible for the behavior:

https://github.com/dehora/nakadi-java/blob/master/nakadi-java-client/src/main/java/nakadi/UriBuilder.java#L63

We are working around by ensuring on our side that we don't provide a base URI which ends with a forward slash, but it would make sense if the client could sanitize this on its own.

**Comment by dehora** [comment]
@a1exsh do you have an example I can look at?

**Comment by a1exsh** [comment]
@dehora sure, just throw in this new test into `UriBuilderTest.java`:
```java
  @Test
  public void buildPathBaseURIWithTrailingSlash() {
    assertEquals("http://example.org/event-types/one",
        UriBuilder.builder("http://example.org/")
            .path("event-types")
            .path("one")
            .buildString());
  }
```
and see it fail:
```
nakadi.UriBuilderTest > buildPathBaseURIWithTrailingSlash FAILED
    org.junit.ComparisonFailure: expected:<http://example.org/[]event-types/one> but was:<http://example.org/[/]event-types/one>
        at org.junit.Assert.assertEquals(Assert.java:115)
        at org.junit.Assert.assertEquals(Assert.java:144)
        at nakadi.UriBuilderTest.buildPathBaseURIWithTrailingSlash(UriBuilderTest.java:20)
```

**Comment by dehora** [comment]
fixed in #331, released as https://github.com/dehora/nakadi-java/releases/tag/0.11.0

**Comment by a1exsh** [comment]
That was fast, thanks! :-)
